vsc/model/vsc_ast.py

```diff
@@ -96,21 +96,60 @@ class Include:
 
 @dataclass
 class Namespace:
+    """
+    Dataclass used to represent VSC Namespace.
+
+    A namespace is a logical grouping of other objects, allowing for separation of datatypes, methods,
+    events, and properties into their own spaces that do not interfere with identically named objects
+    in other namespaces. Namespaces can be nested inside other namespaces to an arbitrary depth, building
+    up a scalable namespace tree. Namespaces can be reused either locally in a single file via YAML anchors,
+    or across YAML files using the includes object. The root of a YAML file is assumed to be a namespaces object,
+    and can contain the keys listed below.
+
+    A namespace example is given below.
+
+    ```yaml
+    namespaces:
+      - name: seats
+        major_version: 1
+        minor_version: 3
+        description: Seat interface and datatypes.
+    ```
+    """
+
     name: str
+    """ Specifies the name of the namespace. """
```

Comment:
@gunnarx how about this style of comments? Just curious about your opinion.

Reply:
In general it makes sense to me if we use it as a (single) source for the documentation and language definition.  
Otherwise it feels like extra stuff to maintain, (and in many cases giving  information that you already know.  I mean, we try to make sure the name of each thing understandable already.   A variable _name_ inside a class called _Namespace_, it's quite obvious that it  _Specifies the name of the namespace_. ".)  
Even despite this, I guess I feel it's OK to write something similar, _iff_ we put it in a single place and it is used as the documentation and official language definition.   

(In many cases I think normally you have to write either nothing, or you have to write a bit more to explain what something really is).

But on the topic of putting this information in one single place - it should work just fine.  The idea shown in #37 could extract the `__doc__` parameter to get the """comments""", and use that to output docs. 

I still think the group would better discuss and agree on some fundamental things first (https://github.com/COVESA/vehicle_service_catalog/pull/41), and then when we have a basic agreement, put it into code.

Also, if we put all of it in the code, it moves the language definition from https://github.com/COVESA/vehicle_service_catalog/ to **vsc-tools**, so that's another decision to agree on in that case.